### ISSUE TITLE
$CFLAGS not respected when generating the .i file

### ISSUE THREAD

**Issue by remexre** [melt-umn/ableC#20]
If `CFLAGS="-I <somewhere>"`, ableC doesn't pick up on that and instead fails the build.

https://github.com/melt-umn/ableC/blob/4bd34c1824d906e5476cfacac7665cfea9f2e34b/edu.umn.cs.melt.ableC/drivers/parseAndPrint/Driver.sv

Can we get environment variables in Silver?

Workaround: call `java -jar ableC.jar whatever.c $CFLAGS`

**Comment by krame505** [comment]
I don't believe gcc automatically includes this, either.  I don't see a reason why we should do otherwise?

**Comment by remexre** [comment]
`make` does, though. Really it's a matter of whether we want to have ableC read the environment variable or if `compile.sh` should pass it in.

**Comment by krame505** [comment]
Yes, `make` has an implicit rule for compiling .c files that uses that make variable (not actually an environment variable, but if it is provided as an environment variable, but make copies the values of environment variable into make variables), which can be overridden.  This should definitely not be something that the compiler worries about.  Also the `CFLAGS` passed to ableC are usually completely different than the ones passed to gcc, anyway.

**Comment by remexre** [comment]
Okay, I'll just change the `compile.sh`'s then. Which `CFLAGS` would you pass to `gcc` but not ableC? In my case, it's just `-I` and `-L` flags.

**Comment by krame505** [comment]
In general, gcc shouldn't need any -I flags when compiling ableC-generated
code (since it is already preprocessed), and ableC shouldn't need any flags
other than -I since it is only running the preprocessor.

Lucas Kramer

**Comment by remexre** [comment]
Should I just try to parse out the `-I` flags then? These are system-wide (eh, user-wide) `CFLAGS`, not ableC-specific.

**Comment by krame505** [comment]
If that is what you are trying to do, you should just be able to update
the C_INCLUDE_PATH environment variable with those include locations, which
should then get read by gcc when it is called by ableC.

Lucas Kramer

**Comment by ericvanwyk** [comment]
Why is this needed if you can just specify it in compile.sh?

These scripts are meant to demonstrate how one would use the extension in
isolation.

We do need some examples that show how to combine multiple extensions and
to use them.

"In practice" I do not think it is bad for people write makefiles for all
of this.  I can see that this can get messy if an extension is going to
want to compile the generated C program with some specific libraries, but
that can be considered a documentation issue, no?

We can discuss this briefly in tomorrow's meeting.

-Eric
